src/notation/qml/MuseScore/NotationScene/internal/NotationSwitchPanel.qml

```diff
@@ -61,57 +62,113 @@ Rectangle {
         order: 1
     }
 
-    RadioButtonGroup {
-        id: notationsView
+    RowLayout {
+        anchors.fill: parent
+        spacing: 0
 
-        anchors.top: parent.top
-        anchors.left: parent.left
-        anchors.bottom: parent.bottom
+        RadioButtonGroup {
+            id: notationsView
 
-        width: Math.min(contentWidth, parent.width)
+            // - 1: don't need to see the right separator of the rightmost tab
+            readonly property bool needsScrollArrowButtons: contentWidth - 1 > root.width
 
-        model: notationSwitchModel
-        currentIndex: 0
-        spacing: 0
+            Layout.fillWidth: true
+            Layout.fillHeight: true
+
+            ScrollBar.horizontal: undefined
+
+            model: notationSwitchModel
+            currentIndex: 0
+            spacing: 0
+
+            clip: true
+            interactive: true
+
+            function scrollLeft() {
+                let leftmostTab = indexAt(contentX, 0)
+                let newLeftmostTab = Math.max(0, leftmostTab - 1)
+                positionViewAtIndex(newLeftmostTab, ListView.Contain)
+            }
 
-        interactive: width < contentWidth
+            function scrollRight() {
+                let rightmostTab = indexAt(contentX + width - 1, 0)
+                let newRightmostTab = Math.min(count - 1, rightmostTab + 1)
+                positionViewAtIndex(newRightmostTab, ListView.Contain)
+            }
+
+            delegate: NotationSwitchButton {
+                id: button
 
-        delegate: NotationSwitchButton {
-            id: button
+                navigation.name: "NotationTab" + index
+                navigation.panel: navPanel
+                navigation.row: index * 10  + 1 // * 10 - for close button
+                navigation.accessible.name: text + (needSave ? (" " + qsTrc("notation", "Not saved")) : "")
 
-            navigation.name: "NotationTab" + index
-            navigation.panel: navPanel
-            navigation.row: index * 10  + 1 // * 10 - for close button
-            navigation.accessible.name: text + (needSave ? (" " + qsTrc("notation", "Not saved")) : "")
+                text: model.title
+                needSave: model.needSave
 
-            text: model.title
-            needSave: model.needSave
+                readonly property bool canCloseOtherTabs: notationsView.count > 2
+                                                          || (notationsView.count === 2 && index === 0)
+                readonly property bool canCloseAllTabs: notationsView.count > 1
 
-            ButtonGroup.group: notationsView.radioButtonGroup
-            checked: index === notationsView.currentIndex
+                contextMenuItems: notationSwitchModel.contextMenuItems(canCloseOtherTabs, canCloseAllTabs)
```

Comment:
I'm keeping the calculation of `canCloseOtherTabs` and `canCloseAllTabs` in QML, so that we can make use of QML's property binding, and `notationSwitchModel.contextMenuItems` will be called again automatically when those properties change.

Reply:
It is bad practice to rely on such dependence.
Let's move the logic to the model

For example 
add contextMenuRequested signal for NotationSwitchButton
on handler get the context menu items from model like notationSwitchModel.contextMenuItems(index)
(canCloseOtherTabs and canCloseAllTabs should be in the model)

OR (and maybe better than previous version)
add contextMenuItems as a role for item in the model